SIT.Manager.Avalonia/ViewModels/ToolsPageViewModel.cs

```diff
@@ -217,4 +236,115 @@ private async Task ClearCache()
+                        {
+                            sb.AppendLine($"Network Interface: {networkInterface.Name}");
+                            sb.AppendLine($"Interface Type: {networkInterface.NetworkInterfaceType.ToString()}");
+                            sb.AppendLine($"Address: {ip.Address}\n");
+                        }
+                    }
+                }
+            }
+
+            //TODO: Add system hardware information
+
+            DiagnosticData.Add(new("Diagnostics.log", sb.ToString()));
+        }
+
+        if (results.IncludeHttpJson)
+        {
+            FileInfo httpJsonFile = new(Path.Combine(_configService.Config.AkiServerPath, "Aki_Data", "Server", "configs", "http.json"));
+            if (httpJsonFile.Exists)
+                DiagnosticData.Add(new(httpJsonFile.Name, await File.ReadAllTextAsync(httpJsonFile.FullName)));
+        }
+
+        //I hate doing it this way but at least icanhazip is owned by cloudlfare
+        HttpResponseMessage resp = await _httpClient.GetAsync("https://ipv4.icanhazip.com/");
+        string externalAddress = await resp.Content.ReadAsStringAsync();
+
+        List<Tuple<string, string>> CleanLogFiles = new(DiagnosticData.Count);
+        foreach (Tuple<string, string> diagnosticFile in DiagnosticData)
+        {
+            CleanLogFiles.Add(new(diagnosticFile.Item1, diagnosticFile.Item2.Replace(externalAddress.Trim(), "REACTED")));
+        }
+
+        Stream? fileStream;
+        Window? mainWindow = (App.Current.ApplicationLifetime as IClassicDesktopStyleApplicationLifetime)?.MainWindow;
+        if (mainWindow != null)
+        {
+            var pickedPath = await mainWindow.StorageProvider.SaveFilePickerAsync(new FilePickerSaveOptions
+            {
+                Title = _localizationService.TranslateSource("ToolsDiagnosticReportSaveFileTitle"),
+                SuggestedFileName = "diagnostics",
+                DefaultExtension = "zip"
+            });
+            if (pickedPath != null)
+            {
+                fileStream = await pickedPath.OpenWriteAsync();
+            }
+            else
+            {
+                return;
+            }
+        }
+        else
+        {
+            fileStream = File.OpenWrite(Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "Downloads"));
+        }
+
+        using (ZipArchive zipArchive = new(fileStream, ZipArchiveMode.Create, true))
+        {
+            foreach (Tuple<string, string> entryData in CleanLogFiles)
+            {
+                var entry = zipArchive.CreateEntry(entryData.Item1);
+                using (Stream entryStream = entry.Open())
+                using (StreamWriter sw = new(entryStream))
+                {
+                    await sw.WriteAsync(entryData.Item2);
+                    await sw.FlushAsync();
+                }
+            }
+        }
+        await fileStream.DisposeAsync();
```

Comment:
Should fileStream not be as part of a using so that way it is always disposed of regardless what happens in here?